aws_lambda_decorators/validators.py

```diff
@@ -337,3 +337,44 @@ def validate(self, value=None):
             return False
         else:
             return True
+
+
+class CurrencyValidator(Validator):
+    """Validation rule to check if a string is a valid currency according to ISO 4217 Currency Code."""
+    ERROR_MESSAGE = "'{value}' is not a valid currency code."
+
+    def __init__(self, error_message=None):
+        """
+        Checks if a string is a valid currency based on ISO 4217
+
+        Args:
+            error_message (str): A custom error message to output if validation fails
+        """
+        super().__init__(error_message)
+
+    @staticmethod
+    def validate(value=None):
+        """
+        Check if a string is a valid currency based on ISO 4217
+
+        Args:
+            value (str): value to validate against a ISO 4217
+        """
+
+        if value is None:
+            return True
+
+        currencies = ["AED", "AFN", "ALL", "AMD", "ANG", "AOA", "ARS", "AUD", "AWG", "AZN", "BAM", "BBD", "BDT", "BGN",
+                      "BHD", "BIF", "BMD", "BND", "BOB", "BRL", "BSD", "BTN", "BWP", "BYN", "BZD", "CAD", "CDF", "CHF",
+                      "CLP", "CNY", "COP", "CRC", "CUC", "CUP", "CVE", "CZK", "DJF", "DKK", "DOP", "DZD", "EGP", "ERN",
+                      "ETB", "EUR", "FJD", "FKP", "GBP", "GEL", "GGP", "GHS", "GIP", "GMD", "GNF", "GTQ", "GYD", "HKD",
+                      "HNL", "HRK", "HTG", "HUF", "IDR", "ILS", "IMP", "INR", "IQD", "IRR", "ISK", "JEP", "JMD", "JOD",
+                      "JPY", "KES", "KGS", "KHR", "KMF", "KPW", "KRW", "KWD", "KYD", "KZT", "LAK", "LBP", "LKR", "LRD",
+                      "LSL", "LYD", "MAD", "MDL", "MGA", "MKD", "MMK", "MNT", "MOP", "MRU", "MUR", "MVR", "MWK", "MXN",
+                      "MYR", "MZN", "NAD", "NGN", "NIO", "NOK", "NPR", "NZD", "OMR", "PAB", "PEN", "PGK", "PHP", "PKR",
+                      "PLN", "PYG", "QAR", "RON", "RSD", "RUB", "RWF", "SAR", "SBD", "SCR", "SDG", "SEK", "SGD", "SHP",
+                      "SLL", "SOS", "SPL", "SRD", "STN", "SVC", "SYP", "SZL", "THB", "TJS", "TMT", "TND", "TOP", "TRY",
+                      "TTD", "TVD", "TWD", "TZS", "UAH", "UGX", "USD", "UYU", "UZS", "VEF", "VND", "VUV", "WST", "XAF",
+                      "XCD", "XDR", "XOF", "XPF", "YER", "ZAR", "ZMW", "ZWD"]
```

Comment:
Probably better as a global so it's not instantiated every time this is called.
Also, since we're doing an `in` check a `set` woule be a better data type than a list.

Reply:
I'll move it to a set and set it as global

Reply:
@jamsidedown this is addressed now. Happy to resolve?

Reply:
I'll resolve it. Looks good